### ISSUE TITLE
Setup Datadog Monitor for `celery.lag` metric

### ISSUE THREAD

**Issue by glogiotatidis** [mozilla/kitsune#3041]
Part of #3015

**Comment by pmac** [comment]
I believe this may have helped catch the [recent issues we had with email notifications](https://bugzilla.mozilla.org/show_bug.cgi?id=1472483). We should bump priority.

**Comment by jgmize** [comment]
As noted in https://github.com/mozilla/kitsune/issues/3262#issuecomment-402326478 there is an existing DMS alert for celery lag.

**Comment by pmac** [comment]
Yes, but DMS errors aren't exactly monitored and don't send alerts to devs and SREs. Perhaps we can have certain DMS alerts like this one bubble up if more than one is received?

**Comment by glogiotatidis** [comment]
A Datadog monitor could help us identify lag increase which may go unnoticed by the 15' DMS interval.

**Comment by glogiotatidis** [comment]
> FYI I just restarted the celery workers due to DMS emails again, which is a recurring issue. I think it may make sense to implement automated restarts until the root cause of the workers getting hung up can be identified.

Submitted PR https://github.com/mozilla/kitsune/pull/3270 to force restart of workers earlier.

**Comment by glogiotatidis** [comment]
I'm working on setting up [the monitor](https://app.datadoghq.com/monitors#5579780/edit).

We have to combine the data from `oregon_a` and `oregon_b` which supposedly datadog can do but I can't get it to work. Emailed support to get some help.

**Comment by glogiotatidis** [comment]
After talking with Datadog support we found a way to combine the two metrics (*) into one Monitor:

`default(max:sumo_prod.oregon_a.celery.lag{*}, 0) + default(max:sumo_prod.oregon_b.celery.lag{*}, 0)`

This combines the two metrics with addition and uses default value of `0` when there's no value. The reason we're using default is that due to the nature of the metric: The task it's either executed by a worker `oregon_a` or a worker in `oregon_b`. This means that when `oregon_a` reports, `oregon_b` doesn't and the other way around. By using a default value we can make this Monitor work and not complain about missing data.

I added useful information on how to restart the celery workers in the metric notification email. I'll continue to monitor the configuration for one week before adding the team to the recipients list to avoid false positives.

(*) Celery lag measurement happens with a Celery task. Since we're running celery workers on two different clusters we have `snippets_prod.oregon_a.celery.lag` and `snippets_prod.oregon_b.celery.lag` which both have to be taken into account to eval celery lag.

**Comment by glogiotatidis** [comment]
I'm going to remove myself from this one. I've too many things on my plate atm and this is probably not a priority since we already have DMS in place.

**Comment by bensternthal** [comment]
Closing this as we haven't used data dog in a looooong time.
